static/src/javascripts/projects/common/modules/discussion/comment-count.js

```diff
@@ -0,0 +1,141 @@
+// @flow
+import fastdom from 'lib/fastdom-promise';
+import fetchJSON from 'lib/fetch-json';
+import { integerCommas } from 'lib/formatters';
+import mediator from 'lib/mediator';
+import { inlineSvg } from 'common/views/svgs';
+
+type IndexedElements = { string: Array<HTMLElement> };
+
+const ATTRIBUTE_NAME: string = 'data-discussion-id';
+const COUNT_URL: string = '/discussion/comment-counts.json?shortUrls=';
+
+const getTemplate = (
+    vals: { url: string, icon: string, count: string },
+    type: string
+): string => {
+    const { url, icon, count } = vals;
+
+    if (type === 'content') {
+        return `<a href="${url}" data-link-name="Comment count" class="commentcount2 tone-colour">
+                    <h3 class="commentcount2__heading">${icon} <span class ="commentcount2__text u-h">Comments</span></h3>
+                    <span class="commentcount2__value tone-colour js_commentcount_actualvalue">${count}</span>
+                </a>`;
+    }
+
+    if (type === 'contentImmersive') {
+        return `<a href="${url}" data-link-name="Comment count" class="commentcount2 tone-colour">
+                    ${icon}<span class="commentcount__value">${count}</span> Comments
+                </a>`;
+    }
+
+    return `<a class="fc-trail__count fc-trail__count--commentcount" href="${url}" data-link-name="Comment count">${icon} ${count}</a>`;
+};
+
+const getElementsIndexedById = (context: HTMLElement): Promise<any> =>
+    fastdom
+        .read(() => context.querySelectorAll(`[${ATTRIBUTE_NAME}]`))
+        .then(elements => {
+            if (elements.length === 0) {
+                return Promise.resolve();
+            }
+
+            return [
+                ...elements,
+            ].reduce((groupedVals: Object, el: HTMLElement): Object => {
+                const attrVal = el.getAttribute(ATTRIBUTE_NAME);
+
+                if (!groupedVals[attrVal]) {
+                    groupedVals[attrVal] = [];
+                }
+
+                groupedVals[attrVal].push(el);
+
+                return groupedVals;
+            }, {});
+        });
+
+const getContentIds = (indexedElements: IndexedElements): string =>
+    Object.keys(indexedElements).sort().join(',');
+
+const getContentUrl = (el: HTMLElement): string => {
+    const a = el.getElementsByTagName('a')[0];
+
+    return `${a ? a.pathname : ''}#comments`;
+};
+
+const updateElement = (el: HTMLElement, count: number): Promise<void> => {
+    const url = el.getAttribute('data-discussion-url') || getContentUrl(el);
+
+    if (el.getAttribute('data-discussion-closed') === 'true' && count === 0) {
+        // Discussion is closed and had no comments, we don't want to show a comment count
+        return Promise.resolve();
+    }
+
+    const format = el.getAttribute('data-commentcount-format') || '';
```

Comment:
Since there are 3 `data-` getters: what do you think about using `el.dataset` once instead with an object spread?

Reply:
I originally converted these to use dataset, but found that `dataset` wasn't supported when testing with Jest/JsDom. I looked into ways around this but didn't come up with anything satisfactory so I converted them back t use `getAttribute`.

Reply:
Oh, yeah, I remember. I really think we shouldn't limit ourselves by the test framework, but patch jest in this case. What do you [think about this](https://github.com/guardian/frontend/blob/master/static/src/javascripts/projects/common/modules/ui/tabs.spec.js#L14)?

Reply:
I definitely agree, it's disappointing to be constricted in our src because of the test framework! 

I looked at your suggestion. I'm unsure how I'd use this though? `dataset` will at least no longer be undefined, however the dataset property lookups will still be `undefined` as `dataset` still won't contain the properties that map to the data attributes in the HTML. 

If I were testing a piece of code that queried `dataset` once I'd set the property in `dataset` from within the test spec however this code iterates across a list elements getting the `dataset` of each, so it'd need to return a different value per element to work as expected.

Reply:
I thought, you'd set `dataset` on the prototype of HTMLElement and therefore you could later assign `myEl.datalist.wat = 'myValue'` to mock the element for the test ...

Reply:
@gustavpursche I've updated to use dataset following your suggestion!